healthcheck/healthcheck.go

```diff
@@ -219,6 +251,15 @@ func (o *Observation) retryCheck(quit chan struct{},
 		var err error
 		select {
 		case err = <-o.Check():
+			// If our error is nil, we have passed our health check,
+			// so we'll invoke our success callback if defined and
+			// then exit.
+			if err == nil {
+				// Invoke the success callback.
+				o.OnSuccess()
```

Comment:
Since a caller can set the callback to `nil`, I think we should do a quick nil check here (and for the `OnFailure` as well) to avoid any panics.

Reply:
Or, the alternative would be to remove the initial assignment of the no-op function and check for nil after applying the functional option. Then if any callback isn't set, we assign the no-op.

Reply:
Ooh, I like that idea. That way we ensure we default to no-op regardless of what callers set and we keep the readability gains of not having to do the nil checks here.

Reply:
✅